streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -85,6 +85,21 @@ public void initializeMetadata() {}
     public boolean initializeStateStores() {
         log.trace("Initializing state stores");
         registerStateStores();
+
+        if (eosEnabled) {
+            final Set<TopicPartition> partitionsToReinitialize = new HashSet<>();
+            for (final TopicPartition partition : stateMgr.checkpointed().keySet()) {
```

Comment:
I'm sure this works, but something seems odd about listing _checkpointed_ partitions to find ones that are _not checkpointed_. Maybe we need to encapsulate this search, or provide a more semantically appropriate "list all changelog partitions" method?

Reply:
Yeah, the fix is mainly for illustrating purpose atm. The problem is really, that the checkpoint file get's deleted for EOS before we reach this point in the code and thus we need to "hack" around it -- it would be cleaner if we could just check if a checkpoint exists of not...

Reply:
I think `checkpointed` is just misleadingly named -- it's really more like checkpoint_able_ offsets IIUC

Reply:
Not sure how to proceed? The naming issue might be a general problem, that we might want to fix in trunk/2.6? Is is worth to fix in 2.5?

Reply:
I'm fine with leaving the poorly-named method as is, since it's fixed in trunk. But maybe you could leave a comment explaining why this seemingly contradictory set of logic actually makes sense (it took me more than a second to understand this at least)